integrations/coinbase/src/main/java/org/dash/wallet/integration/coinbase_integration/network/RemoteDataSource.kt

```diff
@@ -16,55 +16,65 @@
 import org.dash.wallet.common.Configuration
+import org.dash.wallet.integration.coinbase_integration.CoinbaseConstants
+import org.dash.wallet.integration.coinbase_integration.repository.remote.CustomCacheInterceptor
 import org.dash.wallet.integration.coinbase_integration.repository.remote.HeadersInterceptor
 import org.dash.wallet.integration.coinbase_integration.repository.remote.TokenAuthenticator
 import org.dash.wallet.integration.coinbase_integration.service.CloseCoinbasePortalBroadcaster
 import org.dash.wallet.integration.coinbase_integration.service.CoinBaseTokenRefreshApi
+import org.dash.wallet.integration.coinbase_integration.utils.CoinbaseConfig
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
+import java.io.File
 import javax.inject.Inject
 import kotlin.time.Duration.Companion.seconds
 import kotlin.time.toJavaDuration
 
 class RemoteDataSource @Inject constructor(
+    @ApplicationContext private val context: Context,
     private val userPreferences: Configuration,
+    private val config: CoinbaseConfig,
     private val broadcaster: CloseCoinbasePortalBroadcaster
 ) {
 
-    companion object {
-        private const val BASE_URL = "https://api.coinbase.com/"
-    }
-
-    fun <Api> buildApi(api: Class<Api>, ): Api {
-        val authenticator = TokenAuthenticator(buildTokenApi(), userPreferences, broadcaster)
+    fun <Api> buildApi(api: Class<Api>): Api {
+        val authenticator = TokenAuthenticator(buildTokenApi(), userPreferences, config, broadcaster)
         return Retrofit.Builder()
-            .baseUrl(BASE_URL)
-            .client(getRetrofitClient(authenticator))
+            .baseUrl(CoinbaseConstants.BASE_URL)
+            .client(getOkHttpClient(authenticator))
             .addConverterFactory(GsonConverterFactory.create())
             .build()
             .create(api)
     }
 
     private fun buildTokenApi(): CoinBaseTokenRefreshApi {
         return Retrofit.Builder()
-            .baseUrl(BASE_URL)
-            .client(getRetrofitClient())
+            .baseUrl(CoinbaseConstants.BASE_URL)
+            .client(getOkHttpClient())
             .addConverterFactory(GsonConverterFactory.create())
             .build()
             .create(CoinBaseTokenRefreshApi::class.java)
     }
 
-    private fun getRetrofitClient(authenticator: Authenticator? = null): OkHttpClient {
+    private fun getOkHttpClient(authenticator: Authenticator? = null): OkHttpClient {
         return OkHttpClient.Builder()
             .addInterceptor(HeadersInterceptor(userPreferences))
+            .addInterceptor(CustomCacheInterceptor(context, config))
             .connectTimeout(20.seconds.toJavaDuration())
             .callTimeout(20.seconds.toJavaDuration())
             .readTimeout(20.seconds.toJavaDuration())
+            .cache(Cache(
+                directory = CoinbaseConstants.getCacheDir(context),
+                maxSize = 10L * 1024L * 1024L // 10 MB
+            ))
```

Comment:
The built-in cache functionality won't help us with the base_ids (for that we'll use `CustomCacheInterceptor`), but Coinbase APIs provide caching values like `wtag` which okHttp can take advantage of. This should improve networking for the Coinbase feature in general.